lib/spack/spack/build_systems/nmake.py

```diff
@@ -77,7 +77,11 @@ def ignore_quotes(self):
     @property
     def build_directory(self):
         """Return the directory containing the makefile."""
-        return self.pkg.stage.source_path if not self.makefile_root else self.makefile_root
+        return (
+            fs.compute_sfn(self.pkg.stage.source_path)
+            if not self.makefile_root
```

Comment:
It looks like this class deifnes `makefile_root`, so would it ever be falsey?

Reply:
If a package overrides it to be an empty string